building/optee_with_rust.rst

```diff
@@ -0,0 +1,100 @@
+.. _optee_with_rust:
+
+################ 
+OP-TEE with Rust
+################
+
+This document describes how to build OP-TEE client and trusted applications
+written in `Rust <https://www.rust-lang.org>`_ with `Teaclave TrustZone SDK
+<https://github.com/apache/incubator-teaclave-trustzone-sdk>`_.
+
+Clone OP-TEE repo 
+*****************
+
+Currently, Teaclave TrustZone SDK is compatible with QEMUv8 (aarch64).
+
+Before building examples written with Teaclave TrustZone SDK, you should 
+clone the OP-TEE repo first. For QEMUv8, run:
+
+.. code-block:: bash
+
+    $ mkdir YOUR_OPTEE_DIR && cd YOUR_OPTEE_DIR
+    $ repo init -u https://github.com/OP-TEE/manifest.git -m qemu_v8.xml 
+    $ repo sync
+
+The source code of Teaclave TrustZone SDK is located in 
+``YOUR_OPTEE_DIR/optee_rust/`` containing a set of examples written in Rust
+using the SDK.
+
+For more information about building OP-TEE using QEMUv8, see `run OP-TEE using
+QEMU
+<https://optee.readthedocs.io/en/latest/building/devices/qemu.html#qemu-v8>`_ .
+
+Compile Rust examples 
+*********************
+
+To build Rust examples in ``optee_rust/examples/`` and install them in
+Buildroot, run:
+
+.. code-block:: bash
+
+    $ (cd build && make OPTEE_RUST_ENABLE=y CFG_TEE_RAM_VA_SIZE=0x00300000)
+
+Then start QEMUv8:
+
+.. code-block:: bash
+
+    $ (cd build && make run-only)
+
+.. hint::
+
+    Note that if you are under the environment without GUI, you can use
+    ``soc_term`` instead.
+
+    Access to normal world terminal:
+
+    .. code-block:: bash
+
+       $ ./build/soc_term.py 54320
+
+    Access to secure world terminal:
+
+    .. code-block:: bash
+
+       $ ./build/soc_term.py 54321
+
+    Run QEMU:
+
+    .. code-block:: bash
+
+       $ (cd build && make run-only)
+
+To differentiate from `optee_examples
+<https://optee.readthedocs.io/en/latest/building/gits/optee_examples/optee_examples.html#>`_
+, OP-TEE Rust examples are not prefixed with `optee_example_` but suffixed with
+``-rs``. All of them are located in separate directories in
+``YOUR_OPTEE_DIR/optee_rust/examples``. More description about Rust examples can
```

Comment:
I find the references a bit confusing between source filetrees and generated/installed binary file.
I would remove the sentence "All of them are located in separate directories in
``YOUR_OPTEE_DIR/optee_rust/examples``" since this path is already mentioned above.

Reply:
Updated. Thanks!